bin/test-lxd-block-storage

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+set -eux
+
+cleanup() {
+    echo ""
+    if [ "${FAIL}" = "1" ]; then
+        echo "Test failed"
+        exit 1
+    fi
+
+    echo "Test passed"
+    exit 0
+}
+
+poolDriverList="${1:-dir btrfs lvm zfs ceph}"
+FAIL=1
+trap cleanup EXIT HUP INT TERM
+
+# Wait for snapd seeding
+sleep 1m
+
+# Configure to use the proxy
+curl -s http://canonical-lxd.stgraber.org/config/snapd.sh | sh
+
+# Configure for ceph use
+curl -s http://canonical-lxd.stgraber.org/config/ceph.sh | sh
+
+# Install LXD
+while :; do
+    [ ! -e /usr/bin/lxd ] && break
+    apt remove --purge lxd lxd-client --yes && break
+done
+snap remove lxd || true
+snap install lxd --channel=latest/edge
+snap install jq
+lxd waitready --timeout=300
+
+# Configure LXD
+lxc network create lxdbr0
+lxc profile device add default eth0 nic network=lxdbr0
+
+poolName="vmpool$$"
+
+for poolDriver in $poolDriverList
+do
+        echo "==> Create storage pool using driver ${poolDriver}"
+        if [ "${poolDriver}" = "dir" ]; then
+                lxc storage create "${poolName}" "${poolDriver}" volume.size=5GB
```

Comment:
Any particular reason for 5GB in this section, seems like for non-optimized block volumes this will just cause the tests to run slower compared to, say, 50MB? Is there a practical minimum size?

Reply:
Yes, I went with 5GB because I got this error when creating the instance `Failed instance creation: Failed creating instance from image: Source image size (4294967296) exceeds specified volume size (50000000)`

Reply:
done